### ISSUE TITLE
SDDM/Plasma 5.x Screen Refresh Issue On Lock->Login

### ISSUE THREAD

**Issue by BrokenCanoe** [sddm/sddm#652]
When my screen locks after timeout, on returning to log back in, the screen doesn't refresh. So essentially you have visibility of the mouse cursor, cannot select the password field, but you can can type in your password, although nothing happens on the screen. 

If you however switch to another tty, for example Ctrl+f2 and then back to Ctrl+f1, the login screen will suddenly refresh and show your password has been entered, then log back in without any further input.

This seems to happen on both the open-source AMD drivers, and proprietary.
Has anyone else experienced anything similar?

==
--
Graphics: Card: Advanced Micro Devices [AMD/ATI] Kaveri [Radeon R5 Graphics]
Display Server: X.Org 1.17.4 driver: radeon Resolution: 1680x1050@59.88hz
GLX Renderer: Gallium 0.4 on AMD KAVERI (DRM 2.43.0, LLVM 3.8.0) GLX Version: 3.0 Mesa 11.2.2
--
==
Spec : 
KDE Plasma Version : 5.6.5
KDE Frameworks : 5.23.0
Qt : 5.6.1
Kernel : 4.4.13-1-MANJARO
Proc : 2xAMD A6-7400K Radeon R5 (2C+4G) Kaveri

**Comment by yaomtc** [comment]
I've been having an issue very similar to this, might be the same. When I first boot into Arch, and SDDM loads up, I can't see it, as it's just a black screen. No cursor or anything. If I switch to another tty, and then back again, I can see everything. Posted a couple days ago: https://bbs.archlinux.org/viewtopic.php?id=215750

**Comment by Ukalnins** [comment]
Exactly the same case. Screen blanks out until ctrl+alt+f2 -> ctrl+alt+f1 . After wake up, system is slow to respond for some time, there is some hang time until I can input password, after that system runs normal. Running Arch on Intel . Kernel 4.4.16-1-lts.

**Comment by Ukalnins** [comment]
At the time of trying to unlock screen, got this in journalctl:

```
aug 16 09:33:51 kdeinit5[1180]: powerdevil: Can't contact ck
aug 16 09:33:51 kdeinit5[1180]: powerdevil: set screen brightness value:  5
aug 16 09:33:51 dbus[406]: [system] Activating service name='org.kde.powerdevil.backlighthelper' (using servicehelper)
aug 16 09:33:51 dbus[406]: [system] Successfully activated service 'org.kde.powerdevil.backlighthelper'
aug 16 09:33:51 kdeinit5[1180]: powerdevil: Udev device changed "/sys/devices/pci0000:00/0000:00:02.0/backlight/acpi_video0" "/sys/devices/pci0000:00/0000:00:02.0/backlight/acpi_video0"
aug 16 09:34:04 kernel: [drm] stuck on render ring
aug 16 09:34:04 kernel: [drm] GPU HANG: ecode 7:0:0x87c3ffff, in Xorg [654], reason: Ring hung, action: reset
aug 16 09:34:04 kernel: drm/i915: Resetting chip after gpu hang
aug 16 09:34:05 xembedsniproxy[1291]: kde.xembedsniproxy: Scaling pixmap of window 39845917 "skype" from w*h 22 22
aug 16 09:34:05 kdeinit5[1180]: powerdevil: ACTIVE SESSION PATH CHANGED: "/org/freedesktop/login1/session/c2"
aug 16 09:34:05 kdeinit5[1180]: powerdevil: Current session is now active
aug 16 09:34:05 ksmserver[1238]: CreateNotify: 125829123
aug 16 09:34:05 kdeinit5[1180]: powerdevil: ACTIVE SESSION PATH CHANGED: "/org/freedesktop/login1/session/c2"
aug 16 09:34:05 kdeinit5[1180]: powerdevil: Current session is now active
aug 16 09:34:05 kcminit[32418]: Initializing  "kcm_input" :  "kcminit_mouse"
aug 16 09:34:05 ksmserver[1238]: CreateNotify: 125829128
aug 16 09:34:05 kwin_x11[1246]: QXcbConnection: XCB error: 3 (BadWindow), sequence: 53095, resource id: 125829128, major code: 18 (ChangeProperty), minor code: 0
```

**Comment by plfiorini** [comment]
@BrokenCanoe @yaomtc Can you configure a different theme for SDDM like maldives and see if you still have problems?

@Ukalnins the log seems from Plasma, there is even ksmserver and kwin_x11 not SDDM, anyway it seems there's a bug in the drm driver which might affect SDDM as well but in that case the problem needs to be reported to the kernel guys.

**Comment by yaomtc** [comment]
@plfiorini Just tried maldives, has the same issue. It also doesn't have keyboard/mouse focus (I do with Breeze) and I have to click, but that's a different issue...

**Comment by notfood** [comment]
I've got the same issue. Black screen until switch TTYs back and forth.

Tried changing themes from Breeze to others with no avail.

I happen to get some errors:
```
-- Logs begin at Tue 2014-04-01 13:56:25 PET, end at Sat 2016-08-20 03:14:19 PET. --
Aug 15 12:59:21 youmu systemd[369]: Reached target Timers.
Aug 15 12:59:21 youmu systemd[369]: Starting D-Bus User Message Bus Socket.
Aug 15 12:59:21 youmu systemd[369]: Reached target Paths.
Aug 15 12:59:21 youmu systemd[369]: Listening on D-Bus User Message Bus Socket.
Aug 15 12:59:21 youmu systemd[369]: Reached target Sockets.
Aug 15 12:59:21 youmu systemd[369]: Reached target Basic System.
Aug 15 12:59:21 youmu systemd[369]: Starting Update XDG user dir configuration...
Aug 15 12:59:21 youmu systemd[369]: Started Update XDG user dir configuration.
Aug 15 12:59:21 youmu systemd[369]: Reached target Default.
Aug 15 12:59:21 youmu systemd[369]: Startup finished in 82ms.
Aug 15 12:59:22 youmu sddm-greeter[376]: Reading from "/usr/share/xsessions/fluxbox.desktop"
Aug 15 12:59:22 youmu sddm-greeter[376]: Reading from "/usr/share/xsessions/openbox-kde.desktop"
Aug 15 12:59:22 youmu sddm-greeter[376]: Reading from "/usr/share/xsessions/openbox.desktop"
Aug 15 12:59:22 youmu sddm-greeter[376]: Reading from "/usr/share/xsessions/plasma.desktop"
Aug 15 12:59:22 youmu sddm-greeter[376]: Connected to the daemon.
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject::installEventFilter(): Cannot filter events for objects in a different thread.
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu systemd[369]: Started D-Bus User Message Bus.
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject: Cannot create children for a parent that is in a different thread.
                                         (Parent is SDDM::GreeterApp(0x7ffc14614e10), parent's thread is QThread(0x22cc840), current thread is QThread(0x269f5a0)
Aug 15 12:59:22 youmu sddm-greeter[376]: QObject::installEventFilter(): Cannot filter events for objects in a different thread.
Aug 15 12:59:23 youmu sddm-greeter[376]: file:///usr/share/sddm/themes/breeze/components/InfoPane.qml:54:22: Unable to assign [undefined] to int
Aug 15 12:59:23 youmu sddm-greeter[376]: Adding view for "DVI-D-0" QRect(0,0 1920x1080)
Aug 15 12:59:23 youmu sddm-greeter[376]: file:///usr/share/sddm/themes/breeze/components/InfoPane.qml:54:22: Unable to assign [undefined] to int
Aug 15 12:59:23 youmu sddm-greeter[376]: Adding view for "DVI-I-0" QRect(1920,312 1024x768)
Aug 15 12:59:23 youmu sddm-greeter[376]: QDBusConnection: name 'org.freedesktop.UDisks2' had owner '' but we thought it was ':1.9'
Aug 15 12:59:23 youmu sddm-greeter[376]: Message received from daemon: Capabilities
Aug 15 12:59:23 youmu sddm-greeter[376]: Message received from daemon: HostName
```

**Comment by yaomtc** [comment]
After a recent update of nvidia and libevdev, this no longer occurs on startup. However, when I log out, I do have a black screen with a cursor until I use the tty switch method.

**Comment by notfood** [comment]
Updated Nvidia 367.35-2 -> 370.23-1 under Archlinux fixed the issue for me.

**Comment by yaomtc** [comment]
Not sure why that happened before, but logout now works correctly here. Problem seems to be completely fixed for me.

**Comment by int-ua** [comment]
Looks like it's fixed for me too after upgrading Intel graphics with their update tool 2.0.2 from 01.org

**Comment by plfiorini** [comment]
Ok guys since this doesn't happen anymore I'm closing the issue.
Thanks for the report.

**Comment by int-ua** [comment]
Just happened again while logging in to Openbox, sorry. Doesn't look like Plasma is to blame.

**Comment by int-ua** [comment]
And again while logging into Plasma.

**Comment by int-ua** [comment]
FYI: it's not happening anymore, thanks.

**Comment by andreldm** [comment]
@parkerlreed thanks for your tip, it worked very well to me, however instead of haveged, I went for [rngd](https://wiki.archlinux.org/index.php/Rng-tools) which is recommend by Arch's wiki.
Just for the record, I'm also running Arch Linux and since last week SDDM started to hang after boot, I always had to switch between tty's twice to make SDDM appear. Reverting to 0.16.0-3 didn't help. Other DMs such as lightdm, gdm and lxdm did not present this issue, but they have some other problems that's why I insisted to stick to SDDM.
